packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -314,50 +302,344 @@ function TabLine(props: TabLineProps) {
+  let tabIndex = useHasTabbableChild(ref) ? undefined : 0;
+
+  return (
+    <Group
+      {...otherProps}
+      ref={ref}
+      aria-labelledby={menuId}
+      tabIndex={tabIndex}
+      style={UNSAFE_style}
+      className={UNSAFE_className + tabPanel(null, props.styles)} />
+  );
+}
+
+function isAllTabsDisabled<T>(collection: Collection<Node<T>> | undefined, disabledKeys: Set<Key>) {
+  let testKey: Key | null = null;
+  if (collection && collection.size > 0) {
+    testKey = collection.getFirstKey();
+
+    let index = 0;
+    while (testKey && index < collection.size) {
+      // We have to check if the item in the collection has a key in disabledKeys or has the isDisabled prop set directly on it
+      if (!disabledKeys.has(testKey) && !collection.getItem(testKey)?.props?.isDisabled) {
+        return false;
+      }
+
+      testKey = collection.getKeyAfter(testKey);
+      index++;
+    }
+    return true;
+  }
+  return false;
+}
+
+let HiddenTabs = function (props: {
+  listRef: RefObject<HTMLDivElement | null>,
+  items: Array<Node<any>>,
+  size?: string,
+  density?: 'compact' | 'regular'
+}) {
+  let {listRef, items, size, density} = props;
+
+  return (
+    <div
+      // @ts-ignore
+      inert="true"
+      ref={listRef}
+      className={style({
+        display: '[inherit]',
+        flexDirection: '[inherit]',
+        gap: '[inherit]',
+        flexWrap: '[inherit]',
+        position: 'absolute',
+        inset: 0,
+        visibility: 'hidden',
+        overflow: 'hidden',
+        opacity: 0
+      })}>
+      {items.map((item) => {
+        // pull off individual props as an allow list, don't want refs or other props getting through
+        return (
+          <div
+            data-hidden-tab
+            style={item.props.UNSAFE_style}
+            key={item.key}
+            className={item.props.className({size, density})}>
+            {item.props.children({size, density})}
+          </div>
+        );
+      })}
+    </div>
+  );
+};
+
+let TabsMenu = (props: {items: Array<Node<any>>, onSelectionChange: TabsProps['onSelectionChange']} & TabsProps) => {
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/s2');
+  let {id, items} = props;
+  let {density, onSelectionChange, selectedKey, isDisabled, disabledKeys, pickerRef, labelBehavior} = useContext(InternalTabsContext);
+  let state = useContext(TabListStateContext);
+  let allKeysDisabled = useMemo(() => {
+    return isAllTabsDisabled(state?.collection, disabledKeys ? new Set(disabledKeys) : new Set());
+  }, [state?.collection, disabledKeys]);
+
+  return (
+    <div
+      className={style({
+        display: 'flex',
+        flexShrink: 0,
+        alignItems: 'center',
+        height: {
+          density: {
+            compact: 32,
+            regular: 48
+          }
+        }})({density})}>
+      <Picker
+        id={id}
+        aria-label={props['aria-label'] ?? stringFormatter.format('tabs.selectorLabel')}
```

Comment:
Re-added a default here since aria-label isn't required for Tabs, but it is required for a picker that has no visible label.

Reply:
hmm should it be required for Tabs too?

Reply:
That's a possible solution. We haven't required that in the past, and there's nothing on the aria patterns to indicate we should.
That said, it seems like a nice thing to do for users.